packages/core/src/Cardano/types/Address.ts

```diff
@@ -1,23 +1,1259 @@
+    }
+
+    return address;
+  }
+
+  /**
+   * Gets the byte array representation of this address.
+   */
+  // eslint-disable-next-line complexity
+  toBytes(): HexBlob {
+    let cborData;
+    switch (this.#props.type) {
+      case AddressType.BasePaymentKeyStakeKey:
+      case AddressType.BasePaymentScriptStakeKey:
+      case AddressType.BasePaymentKeyStakeScript:
+      case AddressType.BasePaymentScriptStakeScript: {
+        cborData = Address.packBaseAddressParts(this.#props);
+        break;
+      }
+      case AddressType.PointerKey:
+      case AddressType.PointerScript: {
+        cborData = Address.packPointerAddressParts(this.#props);
+        break;
+      }
+      case AddressType.EnterpriseKey:
+      case AddressType.EnterpriseScript: {
+        cborData = Address.packSingleCredAddressParts(this.#props);
+        break;
+      }
+      case AddressType.RewardKey:
+      case AddressType.RewardScript: {
+        cborData = Address.packSingleCredAddressParts(this.#props);
+        break;
+      }
+      case AddressType.Byron: {
+        cborData = Address.packByronAddressParts(this.#props);
+        break;
+      }
+      default:
+        throw new Error('Invalid address'); // Shouldn't happen
+    }
+
+    return HexBlob.fromBytes(cborData);
+  }
+
+  /**
+   * Gets an address instance from a valid Base58 encoded address.
+   */
+  static fromBase58(base58Address: string) {
+    return Address.fromBytes(HexBlob.fromBytes(BaseEncoding.base58.decode(base58Address)));
+  }
+
+  /**
+   * Encodes this address to Base58.
+   *
+   * @throws if is a Shelley address. In principle, it is possible for Shelley address to be encoded in Base58. However,
+   * implementations are discouraged to encode addresses against the convention, as this helps with the goal that lay
+   * users only encounter a single, canonical version of every address. (See CIP-19.)
+   */
+  toBase58(): AddressString {
+    if (this.#props.type !== AddressType.Byron) throw new Error('Only Byron addresses will be encoded in base58');
+
+    return BaseEncoding.base58.encode(Buffer.from(this.toBytes(), 'hex')) as AddressString;
+  }
+
+  /**
+   * Encodes this address to bech32.
+   *
+   * @throws if is a Shelley address. In principle, it is possible for Byron address to be encoded in Bech32. However,
+   * implementations are discouraged to encode addresses against the convention, as this helps with the goal that lay
+   * users only encounter a single, canonical version of every address. (See CIP-19.)
+   */
+  toBech32(): AddressString {
+    const words = BaseEncoding.bech32.toWords(Buffer.from(this.toBytes(), 'hex'));
+
+    if (this.#props.type === AddressType.Byron) throw new Error('Only Shelley addresses will be encoded in bech32');
+    const prefix = Address.getBech32Prefix(this.#props.type, this.#props.networkId!);
+    return BaseEncoding.bech32.encode(prefix, words, MAX_BECH32_LENGTH_LIMIT) as AddressString;
```

Comment:
We should probably remove the existing `bech32` package dependency since we have this now